src/agent/src/main.rs

```diff
@@ -382,6 +402,93 @@ async fn start_sandbox(
     Ok(())
 }
 
+// If we fail to start the AA, ocicrypt won't be able to unwrap keys
+// and container decryption will fail.
+fn init_attestation_components(logger: &Logger) -> Result<()> {
+    // The Attestation Agent will run for the duration of the guest.
+    launch_process(
+        logger,
+        AA_PATH,
+        &vec![
+            "--attestation_sock",
+            AA_ATTESTATION_URI,
+        ],
+        AA_ATTESTATION_SOCKET,
+        DEFAULT_LAUNCH_PROCESS_TIMEOUT,
+    )
+    .map_err(|e| anyhow!("launch_process {} failed: {:?}", AA_PATH, e))?;
+
+    let config_path = OCICRYPT_CONFIG_PATH;
+
+    // The image will need to be encrypted using a keyprovider
+    // that has the same name (at least according to the config).
+    let ocicrypt_config = serde_json::json!({
+        "key-providers": {
+            "attestation-agent":{
+                "ttrpc":CDH_SOCKET
+            }
+        }
+    });
+
+    fs::write(config_path, ocicrypt_config.to_string().as_bytes())?;
+
+    env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", config_path);
```

Comment:
Genuine question - is `OCICRYPT_KEYPROVIDER_CONFIG` envvar needed for the CDH, but not attestation-agent? I did a search in guest-components and couldn't find anything conclusive

Reply:
afaik `OCICRYPT_KEYPROVIDER_CONFIG` is required by ocicrypt-rs for decryption. ocicrypt-rs and image-rs are compiled into kata-agent. So we don't set the env for a spawned child process like AA_KBC_PARAMS for AA or sth like this, but we use it as a global variable in-process.

If image encryption is not part of the initial kata#main-based release, I would recommend to not port that code over and look for other means of configuration.